netadapter/server/grpc/grpcserver.go

```diff
@@ -0,0 +1,71 @@
+package grpc
+
+import (
+	"fmt"
+	"github.com/kaspanet/kaspad/netadapter/server"
+	"github.com/kaspanet/kaspad/wire"
+)
+
+type gRPCServer struct {
+	newConnectionHandler func(connection server.Connection)
+	connections          []server.Connection
+}
+
+// NewGRPCServer creates and starts a gRPC server with the given
+// listening port
+func NewGRPCServer(listeningPort string) (server.Server, error) {
+	fmt.Printf("Listening on 127.0.0.1:%s\n", listeningPort)
+
+	return &gRPCServer{}, nil
+}
+
+// SetNewConnectionHandler sets the new connection handler
+// function for the server
+func (s *gRPCServer) SetNewConnectionHandler(handleNewConnection func(connection server.Connection)) {
+	s.newConnectionHandler = handleNewConnection
+}
+
+// Connect connects to the given address
+// This is part of the Server interface
+func (s *gRPCServer) Connect(address string) (server.Connection, error) {
+	connection := gRPCConnection{}
+	s.newConnectionHandler(&connection)
+	return &connection, nil
+}
+
+// Connections returns a slice of connections the server
+// is currently connected to.
+// This is part of the Server interface
+func (s *gRPCServer) Connections() []server.Connection {
+	return s.connections
+}
+
+func (s *gRPCServer) Close() error {
+	for _, connection := range s.connections {
+		err := connection.Disconnect()
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+type gRPCConnection struct{}
+
+// Send sends the given message through the connection
+// This is part of the Connection interface
+func (c *gRPCConnection) Send(message wire.Message) error {
+	return nil
```

Comment:
Add a TODO comment so it's not lost.
Here in all other unimplemented functions.

Also, preferably panic out of here, unless you need it for testing.